ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/operations/LondonSStoreOperationGasCostTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.vm.operations;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.hyperledger.besu.config.StubGenesisConfigOptions;
+import org.hyperledger.besu.config.experimental.ExperimentalEIPs;
+import org.hyperledger.besu.ethereum.core.Account;
+import org.hyperledger.besu.ethereum.core.Gas;
+import org.hyperledger.besu.ethereum.core.TestCodeExecutor;
+import org.hyperledger.besu.ethereum.mainnet.MainnetProtocolSchedule;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
+import org.hyperledger.besu.ethereum.vm.MessageFrame;
+import org.hyperledger.besu.ethereum.vm.MessageFrame.State;
+
+import org.apache.tuweni.units.bigints.UInt256;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class LondonSStoreOperationGasCostTest {
+
+  private static ProtocolSchedule protocolSchedule;
+
+  @Parameters(name = "Code: {0}, Original: {1}")
+  public static Object[][] scenarios() {
+    // Tests specified in EIP-3539.
+    return new Object[][] {
+      {"0x60006000556000600055", 0, 212, 0},
+      {"0x60006000556001600055", 0, 20112, 0},
+      {"0x60016000556000600055", 0, 20112, 19900},
+      {"0x60016000556002600055", 0, 20112, 0},
+      {"0x60016000556001600055", 0, 20112, 0},
+      {"0x60006000556000600055", 1, 3012, 4800},
```

Comment:
There is no complex logic. Trying to test it I found myself mocking everything. I think it will be more useful to check this part thanks to the reference tests